public/main.js

```diff
@@ -1,198 +1,155 @@
+                    break;
+                    //only displays the completed todos
+                case "complete":
+                    $scope.todos.forEach(function(todo) {
+                        todo.filtered = todo.isComplete ? false : true;
+                    });
+                    break;
             }
-            break;
-            //only displays the active (uncompleted) todos
-        case "active":
-            for (i = 0; i < todoListItems.length; i++) {
-                todoListItems[i].style.display =
-                    todoListItems[i].classList.contains("todo-item-complete") ? "none" : "list-item";
+        };
+
+        /*
+            Rather than replace the whole local list with the fetched list we
+            find the differences in the lists and updated the local list accordingly.
+            This removes flicker on reloads and allows use of angular animations.
+        */
+        $scope.updateTodoList = function(newList) {
+            var type = "";
+            if ($scope.todos.length > newList.length) {
+                type = "deletion";
+            } else if ($scope.todos.length < newList.length) {
+                type = "addition";
+            } else if ($scope.todos.length === newList.length) {
+                type = "update";
             }
-            break;
-            //only displays the completed todos
-        case "complete":
-            for (i = 0; i < todoListItems.length; i++) {
-                todoListItems[i].style.display =
-                    todoListItems[i].classList.contains("todo-item-complete") ? "list-item" : "none";
+            var i = 0;
+            switch (type) {
+                case "deletion":
+                    for (i = 0; i < $scope.todos.length; i++) {
+                        if (!newList[i] || $scope.todos[i].id !== newList[i].id) {
+                            $scope.todos.splice(i, 1);
+                            i--;
+                        }
+                    }
+                    break;
+                case "addition":
+                    for (i = 0; i < newList.length; i++) {
+                        if (!$scope.todos[i] || $scope.todos[i].id !== newList[i].id) {
+                            $scope.todos.splice(i, 0, newList[i]);
+                            i--;
+                        }
+                    }
+                    break;
+                case "update":
+                    for (i = 0; i < $scope.todos.length; i++) {
+                        if (!_.isMatch($scope.todos[i], newList[i])) {
+                            $scope.todos.splice(i, 1, newList[i]);
+                        }
+                    }
+                    break;
             }
-            break;
-    }
-}
-reloadTodoList();
-var timerId = setInterval(reloadTodoList, 30000);
+        };
+
+        angular.element(document).ready(function() {
+            $scope.getTodos();
+        });
+        var pollTimer = setInterval($scope.getTodos, 30000);
+
+    }]);
```

Comment:
In general, you have a lot of logic inside your ToDoController. I would recommend pulling this out and making a ToDoService. Your controller should be a small as possible.